docs/data/data-grid/editing/EditingWithDatePickers.tsx

```diff
@@ -175,21 +176,30 @@ const dateColumnType: GridColTypeDef<Date | string, string> = {
   },
 };
 
+const GridEditDateTextField = styled(TextField)({
+  '& .MuiInputBase-root': {
+    fontSize: 'inherit',
+  },
+});
+
 function GridEditDateCell({
   id,
   field,
   value,
+  colDef,
 }: GridRenderEditCellParams<Date | string | null>) {
   const apiRef = useGridApiContext();
 
+  const Component = colDef.type === 'dateTime' ? DateTimePicker : DatePicker;
+
   const handleChange = (newValue: unknown) => {
     apiRef.current.setEditCellValue({ id, field, value: newValue });
   };
 
   return (
-    <DatePicker
+    <Component
       value={value}
-      renderInput={(params) => <TextField {...params} />}
+      renderInput={(params) => <GridEditDateTextField fullWidth {...params} />}
```

Comment:
I see that we have kept the `TextField` for this demo but shouldn't we have an `InputBase` instead?

It looks strange, in https://deploy-preview-5304--material-ui-x.netlify.app/x/react-data-grid/editing/#usage-with-mui-x-date-pickers, there are two borders, the one from the data grid edit and the one from the textfield.

<img width="228" alt="Screenshot 2022-07-23 at 16 06 29" src="https://user-images.githubusercontent.com/3165635/180608585-a1ff547a-29df-4204-a711-f15caa4bd294.png">